src/front/wgsl/mod.rs

```diff
@@ -1046,12 +1046,10 @@ impl Parser {
                     type_arena,
                     const_arena,
                 )?;
-                lexer.expect(Token::Paren('('))?;
+
+                lexer.open_arguments()?;
                 let mut components = Vec::new();
-                while !lexer.skip(Token::Paren(')')) {
-                    if !components.is_empty() {
-                        lexer.expect(Token::Separator(','))?;
-                    }
+                while components.is_empty() || lexer.next_argument()? {
                     let component = self.parse_const_expression(lexer, type_arena, const_arena)?;
                     components.push(component);
                 }
```

Comment:
This requires at least one argument. The old code accepted zero or more. It seems to me the WGSL grammar requires at least one argument, so perhaps this is deliberate. (But I'm not sure whether to trust the WGSL grammar, since it also doesn't permit the trailing comma at the moment.)

Reply:
It's safe to require at least one right now. Maybe when we get empty structs this will change. I'll leave a comment.